### DISCUSSION TITLE
Make database Project

### DISCUSSION THREAD

**Discussion by mfhum** [Humteria/humteria-backend#18] [General]
At the moment I'm working on the registration and I'm asking myself, if it would make sense to Split Data into Data and Database, as Data Processing and Database as an outlet to the Database, we would kind of have a mvc / mvvm with that
@baltermia your opinion?

**Comment by baltermia** [comment]
sounds good. The Database holds the Models for the DB (since that's necessary for EF). The other Project would contain stuff like DTO Models, object conversion (DTO <-> DB-Models), validation etc. what do you think?

**Comment by mfhum** [comment]
Yes EF Core for the Database Project (Naming not fixed) And DTOs on the Data Project, mapper will be in Data, as it Processes data
Should we create an Issue for that?
Before that we should settle on Naming:
### WebAPI:
Takes and gives calls to the API, also Contains Helpers / Services that build responses based on Data especially Password hashing (could be discussed if better in Data
### Processing:
Contains all the DTOs, Mapper, interfaces and helpers needed for Data Processing
### Database:
Contains all the Models and DB Context, any changes made via SQL are defined in Service

**Comment by baltermia** [comment]
I didn't really like the name `Processing` so I asked ChatGPT for some common naming:
> 1. WebAPI Project: This is the top layer where your controllers are located. It handles incoming HTTP requests, manages routing, and is responsible for interacting with clients. It should be as lightweight as possible and mainly responsible for receiving requests, validating them, and returning responses.
> 
> 2. Service or Application Project: This is the intermediary layer you're talking about. It can be called something like "Application" or "Services." This layer should handle business logic, request validation, and DTO-to-database model mapping. It can also perform other application-specific tasks. It keeps the controllers in the WebAPI project clean and focused on handling HTTP-related concerns.
> 
> 3. Data Project: This is the lower layer responsible for interacting with the database. It contains your database models, Entity Framework Core logic, and any data-related operations.

I like the name `Application` since we can put other concerns like email stuff and so on later and I've heard the term Application Layer a lot before. What do you think?
```
WebAPI
  |
Application
  |
Data
```
---

Or did you think of the Processing Project as a layer *besides* the two, so WebAPI still directly uses the Data project? In that case the Application layer name would make no sense.
```
WebAPI -> Processing
  |           |
Data         <-
```

**Comment by mfhum** [comment]
I like Application, creating an Issue based on this convo

**Comment by mfhum** [comment]
Issue link: https://github.com/Humteria/humteria-backend/issues/19

**Reply by baltermia** [reply]
great
